opamp/observiq/observiq_client.go

```diff
@@ -0,0 +1,186 @@
+	// Create collect client based on URL scheme
+	switch opampURL.Scheme {
+	case "ws", "wss":
+		observiqClient.opampClient = client.NewWebSocket(clientLogger)
+	default:
+		return nil, ErrUnsupportedURL
+	}
+
+	// Add configs to config manager
+	configManager.AddConfig(ManagerConfigName, &opamp.ManagedConfig{
+		ConfigPath: args.ManagerConfigPath,
+		Reload:     managerReload(observiqClient, args.ManagerConfigPath),
+	})
+
+	configManager.AddConfig(CollectorConfigName, &opamp.ManagedConfig{
+		ConfigPath: args.CollectorConfigPath,
+		Reload:     collectorReload(observiqClient, args.CollectorConfigPath),
+	})
+
+	configManager.AddConfig(args.LoggerConfigPath, &opamp.ManagedConfig{
+		ConfigPath: args.LoggerConfigPath,
+		Reload:     loggerReload(observiqClient, args.LoggerConfigPath),
+	})
+
+	return observiqClient, nil
+}
+
+// Connect initiates a connection to the OpAmp server
+func (c *Client) Connect(ctx context.Context) error {
+	// Compose and set the agent description
+	if err := c.opampClient.SetAgentDescription(c.ident.ToAgentDescription()); err != nil {
+		c.logger.Error("Error while setting agent description", zap.Error(err))
+		return err
+	}
+
+	settings := types.StartSettings{
+		OpAMPServerURL:      c.endpoint,
+		AuthorizationHeader: fmt.Sprintf("Secret-Key: %s", c.secretKey),
+		TLSConfig:           nil, // TODO add support for TLS
+		InstanceUid:         c.ident.agentID,
+		Callbacks: types.CallbacksStruct{
+			OnConnectFunc:          c.onConnectHandler,
+			OnConnectFailedFunc:    c.onConnectFailedHandler,
+			OnErrorFunc:            c.onErrorHandler,
+			OnRemoteConfigFunc:     c.onRemoteConfigHandler,
+			GetEffectiveConfigFunc: c.onGetEffectiveConfigHandler,
+			// Unimplemented Handles
+			// SaveRemoteConfigStatusFunc:
+			// OnOpampConnectionSettingsFunc
+			// OnOpampConnectionSettingsAcceptedFunc
+			// OnOwnTelemetryConnectionSettingsFunc
+			// OnOtherConnectionSettingsFunc
+			// OnPackagesAvailableFunc
+			// OnAgentIdentificationFunc
+			// OnCommandFunc
+		},
+	}
+
+	// Start the embedded collector
+	if err := c.collector.Run(ctx); err != nil {
+		return fmt.Errorf("collector failed to start: %w", err)
+	}
+
+	return c.opampClient.Start(ctx, settings)
+}
+
+// Disconnect disconnects from the server
+func (c *Client) Disconnect(ctx context.Context) error {
+	c.collector.Stop()
+	return c.opampClient.Stop(ctx)
+}
+
+// client callbacks
+
+func (c *Client) onConnectHandler() {
+	c.logger.Info("Successfully connected to server")
+}
+
+func (c *Client) onConnectFailedHandler(err error) {
+	c.logger.Error("Failed to connect to server", zap.Error(err))
+}
+
+func (c *Client) onErrorHandler(errResp *protobufs.ServerErrorResponse) {
+	c.logger.Error("Server returned an error response", zap.String("Error", errResp.GetErrorMessage()))
+}
+
+func (c *Client) onRemoteConfigHandler(_ context.Context, remoteConfig *protobufs.AgentRemoteConfig) (*protobufs.EffectiveConfig, bool, error) {
+	c.logger.Debug("Remote config handler")
```

Comment:
minor. seems like this could be info level

Reply:
This could go off a lot I don't want to spam the logs with it. It's more just for us to know what's happening if things are acting weird.